pkg/services/ngalert/store/alertmanager.go

```diff
@@ -132,6 +134,43 @@ func (st *DBstore) UpdateAlertmanagerConfiguration(ctx context.Context, cmd *mod
 	})
 }
 
+// MarkConfigurationAsApplied sets the `applied_at` field of the last config with the given hash to the current UNIX timestamp.
+func (st *DBstore) MarkConfigurationAsApplied(ctx context.Context, cmd *models.MarkConfigurationAsAppliedCmd) error {
+	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+		// Create a map representing the field to be updated and its new value.
+		update := map[string]interface{}{"applied_at": time.Now().UTC().Unix()}
+		rowsAffected, err := sess.Table("alert_configuration_history").
+			Desc("id").
+			Limit(1).
```

Comment:
Idea: What if we marked _all_ rows with the same hash? If it's the same hash, it's the same config! Not sure what others think here.

Reply:
I thought about that, but I decided to update just the latest row for two main reasons:
1. Marking all identical configs as applied at the same time would mean using the same timestamp. We could end up with the same config stored in different points in time but "applied" in the exact same second.
2. We would also have to check for `0` or `NULL` in the the `applied_at` column before the update. Otherwise, we would end up overwriting this field for older configs just because the hash is the same.

I think it would be misleading to fill this column for older rows with the timestamp that's actually for the latest row. The config is the same, but the data is not accurate and it could be misleading both for users (this is supposed to be shown in the UI) and for us in case we have to troubleshoot something related to applied configs.

This would be less of a problem if we were using a boolean, but it would also not be 100% accurate (e.g. suppose we have a config that was valid one year ago but is not anymore).

Reply:
IMO it's similarly not accurate to mark the history entry with most recent ID and that configuration hash. What if the same configuration appears multiple times in history? We don't know that the user actually picked the most recent one right?

Reply:
Unless I'm missing something, yes, the user will always be using the latest config in the history table as I explained in my [comment](https://github.com/grafana/grafana/pull/61330#issuecomment-1387226993).
If a user picks an older config then that config gets re-saved, generating a new record in the history table. So, even if the same config appears multiple times, and even if the applied config is a previous config, the last one is the one in use.